src/app/clusters/device-energy-management-server/device-energy-management-server.cpp

```diff
@@ -226,19 +236,88 @@ void Instance::InvokeCommand(HandlerContext & handlerContext)
                 [this](HandlerContext & ctx, const auto & commandData) { HandleRequestConstraintBasedForecast(ctx, commandData); });
         }
         return;
+    case CancelRequest::Id:
+        if (!HasFeature(Feature::kStartTimeAdjustment) && !HasFeature(Feature::kForecastAdjustment) &&
+            !HasFeature(Feature::kConstraintBasedAdjustment))
+        {
+            handlerContext.mCommandHandler.AddStatus(handlerContext.mRequestPath, Status::UnsupportedCommand);
+        }
+        else
+        {
+            HandleCommand<CancelRequest::DecodableType>(
+                handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleCancelRequest(ctx, commandData); });
+        }
+        return;
     }
 }
 
-void Instance::HandlePowerAdjustRequest(HandlerContext & ctx, const Commands::PowerAdjustRequest::DecodableType & commandData)
+Status Instance::CheckOptOutAllowsRequest(AdjustmentCauseEnum adjustmentCause)
 {
-    int64_t power        = commandData.power;
-    uint32_t durationSec = commandData.duration;
-    bool validArgs       = false;
-    Status status        = Status::Success;
+    OptOutStateEnum optOutState = mDelegate.GetOptOutState();
+
+    if (adjustmentCause >= AdjustmentCauseEnum::kUnknownEnumValue)
```

Comment:
```suggestion
    if (adjustmentCause == AdjustmentCauseEnum::kUnknownEnumValue)
```
Valid values can be larger than kUnknownEnumValue.  What data model decoding guarantees is that all unknown values are coerced to kUnknownEnumValue.